src/useCookie.ts

```diff
@@ -0,0 +1,25 @@
+import { useState, useCallback } from 'react';
+import Cookies from 'js-cookie';
+
+const useCookie = (
+  cookieName: string
+): [string | null, (newValue: string, options?: Cookies.CookieAttributes) => void, () => void] => {
+  const [value, setValue] = useState<string | null>(() => Cookies.get(cookieName) || null);
```

Comment:
Shall we use `undefined` instead of `null`, same as `js-cookie`?

Reply:
I used `null` instead of `undefined` here because we are explicitly setting the value to a nullish value in `deleteCookie`. It is very rare to programmatically assign any value to undefined.

Reply:
Can you set it to `undefined` in `deleteCookie`?

Reply:
Yes I can, but I am not sure if it would be the correct approach. If we have followed that convention all across, I can go ahead and make it `undefined`. Do you want me to go ahead and do that?

Reply:
I was curious if there is a reason it has to be `setValue(null)` in `deleteCookie`, because you were saying: "...because we are explicitly setting the value to a nullish value in `deleteCookie`". Its up to you how to leave it.

Reply:
We do have to set it to `null` so as to cause a rerender since we do not have a listener for cookie change. I think it is better if we use `null` instead of `undefined` here.